src/main/java/seedu/cafectrl/command/BuyIngredientCommand.java

```diff
@@ -61,25 +62,39 @@ private void addIngredient() {
             ingredient = pantry.addIngredientToStock(ingredient.getName(),
                     ingredient.getQty(),
                     ingredient.getUnit());
-            buildBuyIngredientMessage(ingredient, i);
+            ingredients.set(i, ingredient);
+        }
+
+        for (int i = ingredients.size() - 1; i > -1; i--) {
+            Ingredient ingredient = ingredients.get(i);
+            buildBuyIngredientMessage(ingredient);
         }
     }
 
     /**
      * Builds a message about the bought ingredient and appends it to the result message.
      *
      * @param ingredient The Ingredient object to build the message for.
-     * @param index The index of the ingredient in the list.
      */
-    private void buildBuyIngredientMessage(Ingredient ingredient, int index) {
+    private void buildBuyIngredientMessage(Ingredient ingredient) {
+        if (isReapeatedIngredient(ingredient)) {
+            return;
+        }
+        ingredientsToBePrinted.add(ingredient);
         ingredientString += "Ingredient: " + ingredient.getName()
                 + "\t\tQty: " + ingredient.getQty()
-                + ingredient.getUnit();
+                + ingredient.getUnit() + "\n";
+    }
 
-        //append new line if current ingredient is not last
-        if(index < ingredients.size() - ui.OFFSET_LIST_INDEX) {
-            ingredientString += "\n";
+    private boolean isReapeatedIngredient(Ingredient ingredient) {
+        for (Ingredient printedIngredient : ingredientsToBePrinted) {
+            String printedIngredientName = printedIngredient.getName();
+            String ingredientName = ingredient.getName();
+            if (printedIngredientName.equalsIgnoreCase(ingredientName)) {
+                return true;
+            }
         }
+        return false;
     }
```

Comment:
Perhaps you could use my repeated ingredient checking function in the parser cLASS